### DISCUSSION TITLE
Images carousel

### DISCUSSION THREAD

**Discussion by nipnipj** [quarto-dev/quarto-cli#2570] [Feature Requests]
Is there a way within quarto to create an image Slideshow/Carousel? I mean within a blog/website, not creating a presentation.

**Comment by cderv** [comment]
It is not built-in in Quarto but can definitely be done using the tools available. 

Quarto is built on bootstrap so using built-in carousel feature is possible : https://getbootstrap.com/docs/4.0/components/carousel/

Creating an [Extension](https://quarto.org/docs/extensions/) for that would be great! I don't think there is one for now. 

FWIW In quarto-web, we have a carousel at the top here: https://quarto.org/docs/gallery/ but we use R code to make that happens
See source: https://github.com/quarto-dev/quarto-web/tree/main/docs/gallery

**Comment by tomicapretto** [comment]
I've just created this extension for carousels in Quarto: https://github.com/tomicapretto/quarto-carousel.

It's my first time creating a quarto extension, so any feedback would be appreciated.

**Reply by mcanouil** [reply]
Nice.
Feel free to submit it to my extensions listing following the guidelines: https://m.canouil.dev/quarto-extensions/about.html

You can also add it in https://quarto.org/docs/extensions/

**Reply by tomicapretto** [reply]
Will do! Thanks for the quick reply @mcanouil :)

**Reply by mcanouil** [reply]
@tomicapretto If you are interested, I wrote a small blog post: [Lua Best Practices for Quarto Extensions: Building Maintainable and Robust Extensions](https://mickael.canouil.fr/posts/2025-11-06-quarto-extensions-lua/)
You can also look at existing extensions, I would recommend looking at the extensions from James Balamuta (https://github.com/coatless-quarto) and my extensions, which in my opinion are the most structured extensions, but there are many interesting things in others too (JavaScript and CSS tricks, etc.).

**Reply by tomicapretto** [reply]
Looks super helpful, thanks @mcanouil!
